client/post-editor/media-modal/secondary-actions.jsx

```diff
@@ -95,9 +95,11 @@ const MediaModalSecondaryActions = React.createClass( {
 			} );
 		}
 
-		const canDeleteItems = selectedItems.length && every( selectedItems, ( item ) => {
-			return canUserDeleteItem( item, user, site );
-		} );
+		const canDeleteItems = ModalViews.EDIT !== activeView &&
```

Comment:
Since the active view doesn't change a user's capabilities, lets move this down below instead. Maybe something like:
```
if ( ! includes( [ ModalViews.GALLERY, ModalViews.EDIT ], activeView ) && canDeleteItems )
```